src/utils/proxy.js

```diff
@@ -328,20 +329,8 @@ const initializeProxy = function (port, distDir, projectDir) {
   return handlers
 }
 
-const startProxy = async function (settings, addonsUrls, configPath, projectDir) {
-  const functionsServer = settings.functionsPort ? `http://localhost:${settings.functionsPort}` : null
-
-  const proxy = initializeProxy(settings.frameworkPort, settings.dist, projectDir)
-
-  const rewriter = await createRewriter({
-    distDir: settings.dist,
-    jwtSecret: settings.jwtSecret,
-    jwtRoleClaim: settings.jwtRolePath,
-    configPath,
-    projectDir,
-  })
-
-  const server = http.createServer(async function onRequest(req, res) {
+const getRequestListener = ({ proxy, rewriter, settings, addonsUrls, functionsServer }) => {
+  const onRequest = async (req, res) => {
```

Comment:
[sand] What are your thoughts on merging both functions to a single one, then using `onRequest.bind(undefined, { proxy, ... })`?

Reply:
Done https://github.com/netlify/cli/pull/2132/commits/c323df52977c3798da2e93e3b969eb675e08405b